src/main/java/io/getunleash/repository/BackupHandler.java

```diff
@@ -0,0 +1,7 @@
+package io.getunleash.repository;
```

Comment:
I like this abstraction, but I would prefer general cleanup to happen in separate PRs as the number of changes in this PR gets a bit on the higher end in order to provide accurate feedback.